shared/js/background/components/dashboard-messaging.js

```diff
@@ -0,0 +1,92 @@
+import browser from 'webextension-polyfill';
+import { breakageReportForTab, getDisclosureDetails } from '../broken-site-report';
+import { dashboardDataFromTab } from '../classes/privacy-dashboard-data';
+import { registerMessageHandler } from '../message-handlers';
+import { getCurrentTab } from '../utils';
+import { isFireButtonEnabled } from './fire-button';
+
+/**
+ * Message handlers for communication from the dashboard to the extension background.
+ *
+ * Note, additional message handles for toggle reports is in the separate ToggleReports component.
+ */
+export default class DashboardMessaging {
+    /**
+     * @param {{
+     *  settings: import('../settings.js');
+     *  tds: import('./tds').default;
+     *  tabManager: import('../tab-manager.js');
+     * }} args
+     */
+    constructor({ settings, tds, tabManager }) {
+        this.settings = settings;
+        this.tds = tds;
+        this.tabManager = tabManager;
+
+        registerMessageHandler('submitBrokenSiteReport', (report) => this.submitBrokenSiteReport(report));
+        registerMessageHandler('getPrivacyDashboardData', this.getPrivacyDashboardData.bind(this));
+        registerMessageHandler('getBreakageFormOptions', getDisclosureDetails);
+    }
+
+    /**
+     * Only the dashboard sends this message, so we import the types from there.
+     * @param {import('@duckduckgo/privacy-dashboard/schema/__generated__/schema.types').BreakageReportRequest} breakageReport
+     * @param {string} [pixelName]
+     * @param {string} [reportFlow]
+     * @returns {Promise<void>}
+     */
+    async submitBrokenSiteReport(breakageReport, pixelName = 'epbf', reportFlow = undefined) {
+        // wait for config and TDS so we can get etags and config version
+        await Promise.all([this.tds.remoteConfig.allLoadingFinished, this.tds.tds.ready]);
```

Comment:
Is it just this `await Promise.all([this.tds.remoteConfig.allLoadingFinished, this.tds.tds.ready]);` that's resolving the race condition, or are there other changes here that do too? I'm thinking we should split this into two PRs, one for the race condition and one for the refactor.